src/signal-handling.c

```diff
@@ -632,10 +632,14 @@ void jl_critical_error(int sig, int si_code, bt_context_t *context, jl_task_t *c
             sigaddset(&sset, sig);
         pthread_sigmask(SIG_UNBLOCK, &sset, NULL);
 #endif
-        if (si_code)
+        if (si_code != INT_MAX)
             jl_safe_printf("\n[%d] signal %d (%d): %s\n", getpid(), sig, si_code, strsignal(sig));
         else
             jl_safe_printf("\n[%d] signal %d: %s\n", getpid(), sig, strsignal(sig));
+#ifdef _OS_LINUX_
+        if ((sig == SIGBUS || sig == SIGSEGV) && (si_code <= 0))
```

Comment:
```suggestion
        if ((sig == SIGBUS || sig == SIGSEGV) && (si_code <= 0)) # SI_FROMUSER(&si_code)
```

But none of these seem really that interesting here to deserve a special comment when our signal handler is installed but gets one of these sent:
```
/*
 * si_code values
 * Digital reserves positive values for kernel-generated signals.
 */
#define SI_USER         0               /* sent by kill, sigsend, raise */
#define SI_KERNEL       0x80            /* sent by the kernel from somewhere */
#define SI_QUEUE        -1              /* sent by sigqueue */
#define SI_TIMER        -2              /* sent by timer expiration */
#define SI_MESGQ        -3              /* sent by real time mesq state change */
#define SI_ASYNCIO      -4              /* sent by AIO completion */
#define SI_SIGIO        -5              /* sent by queued SIGIO */
#define SI_TKILL        -6              /* sent by tkill system call */
#define SI_DETHREAD     -7              /* sent by execve() killing subsidiary threads */
#define SI_ASYNCNL      -60             /* sent by glibc async name lookup completion */
```